docs/custom_promise_types/rss.py

```diff
@@ -0,0 +1,172 @@
+            if response.ok:
+                return response.content
+            self.log_error(f"Failed to fetch feed from url '{path}'': status code '{response.status_code}'")
+            return None
+
+        # fetch from file
+        try:
+            self.log_verbose(f"Reading feed from file '{path}'")
+            with open(path, 'r') as f:
+                resource = f.read()
+                return resource
+        except Exception as e:
+            self.log_error(f"Failed to open file '{path}' for reading: {e}")
+            return None
+
+
+    def _get_items(self, res):
+        # extract descriptions in /channel/item
+        try:
+            self.log_verbose("Parsing feed")
+            items = []
+            root = ET.fromstring(res)
+            for item in root.findall('./channel/item'):
+                for child in item:
+                    if child.tag == 'description':
+                        items.append(child.text)
+            return items
+        except Exception as e:
+            self.log_error(f"Failed to parse feed: {e}")
+            return None
+
+
+    def _pick_item(self, items, attributes):
+        # Pick newest item as default
+        item = items[0]
+
+        # Select item from feed
+        if "select" in attributes:
+            select = attributes['select']
+            if select == 'random':
+                self.log_verbose("Selecting random item from feed")
+                item = random.choice(items)
+            elif select == 'oldest':
+                self.log_verbose("Selecting oldest item from feed")
+                item = items[len(items) - 1]
+            else:
+                self.log_verbose("Selecting newest item from feed")
+        else:
+            self.log_verbose("Selecting newest item as default")
+        return item
+
+
+    def _write_promiser(self, item, promiser):
+        file_exist = os.path.isfile(promiser)
+
+        if file_exist:
+            try:
+                with open(promiser, 'r') as f:
+                    if f.read() == item:
+                        self.log_verbose(f"File '{promiser}' exists and is up to date, no changes needed")
+                        return Result.KEPT
+            except Exception as e:
+                self.log_error(f"Failed to open file '{promiser}' for reading: {e}")
+                return Result.NOT_KEPT
+
+        try:
+            with open(promiser, 'w') as f:
+                if file_exist:
+                    self.log_info(f"File '{promiser}' exists but contents differ, updating content")
+                else:
+                    self.log_info(f"File '{promiser}' does not exist, creating file")
+                f.write(item)
+                return Result.REPAIRED
+        except Exception as e:
+            self.log_error(f"Failed to open file '{promiser}' for writing: {e}")
+            return Result.NOT_KEPT
+
+
+    def _is_win_file(self, path):
+        regex = re.compile('^[a-zA-Z]:\\\\[\\\\\\S|*\\S]?.*$')
```

Comment:
For regular expressions in Python, it's common to use raw strings (`r""`) so that you can avoid having to use `\\` everywhere.